src/events/mongodb.js

```diff
@@ -1,105 +1,116 @@
+ * @returns {Number} the number of documents that mongodb operation affected
  */
-const getMongodbMetadata = (event) => {
-    const { commandName, reply } = event;
-    let metaData = {};
-    switch (commandName) {
+function getItemsCount(operationName, response) {
+    let itemsCount;
+    switch (operationName) {
     case 'find':
-        if (reply.cursor && Array.isArray(reply.cursor.firstBatch)) {
-            metaData = { items_count: reply.cursor.firstBatch.length };
-        }
+        itemsCount = response.result.cursor.firstBatch.length;
         break;
-    case 'getMore':
-        if (reply.cursor && Array.isArray(reply.cursor.nextBatch)) {
-            metaData = { items_count: reply.cursor.nextBatch.length };
-        }
+    case 'insert':
+    case 'update':
+    case 'delete':
+        itemsCount = response.result.n;
         break;
-    case 'count':
-        if (reply.ok) {
-            metaData = { items_count: reply.n };
-        }
+    case 'getMore':
+        itemsCount = response.cursor.nextBatch.length;
         break;
     default:
         break;
     }
 
-    return metaData;
-};
+    return itemsCount;
+}
 
 /**
- * Extracts connection details from a conenction
- * @param {Object} connId The connectionId object of the instrumented event
- * @return {{port: *, host: *}} extracted connection details
+ * Extracts the relevant arguments from provided arguments
+ * @returns {Object} Object of the extracted arguments
  */
-function getConnectionDetails(connId) {
-    let host;
-    let port;
-    utils.debugLog('Epsagon - MongoDB - inspecting connectionId', connId);
-    if (connId) {
-        if (typeof connId === 'string') {
-            const parts = connId.split(':');
-            if (parts.length && parts[0][0] === '/') {
-                host = 'localhost';
-                [port] = parts;
-            } else {
-                [host, port] = parts;
-            }
-        } else if (connId.domainSocket) {
-            [host, port] = ['localhost', connId.host]; // host for domainSocket is the identifier
-        } else {
-            // eslint-disable-next-line prefer-destructuring
-            host = connId.host;
-            // eslint-disable-next-line prefer-destructuring
-            port = connId.port;
-        }
-    }
-    return { host, port };
+function getArgsFromFunction(...args) {
+    return {
+        server: args[0],
+        namespace: args[1],
+        cmd: args[args.length - 2] === 'getMore' ? {} : args[2],
+        callback: args[args.length - 3],
+        operationName: args[args.length - 2],
+        wrappedFunction: args[args.length - 1],
+    };
 }
 
 /**
- * Hook for mongodb requests starting
- * @param {Object} event The request event
+ * Wrap Mongodb operations call with tracing
+ * @returns {Array} Execiton of the called function
  */
-function onStartHook(event) {
+function internalMongodbOperationWrapper(...args) {
+    let patchedCallback;
```

Comment:
initialize to `callback` otherwise we risk not calling the user callback in an exception in our instrumentation